src/control/cmd/daos/property_ctest.go

```diff
@@ -0,0 +1,125 @@
+//
+// (C) Copyright 2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package main
+
+/*
+#include "util.h"
+
+#include <daos/multihash.h>
+#include <daos/compression.h>
+#include <daos/cipher.h>
+#include <daos/object.h>
+#include <daos/cont_props.h>
+*/
+import "C"
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/dustin/go-humanize"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+)
+
+func testProperty_EcCellSize(testRunner *testing.T) {
+	for testingName, testData := range map[string]struct {
+		SizeStr    string
+		EntryBytes uint64
+	}{
+		"Human Minimal Entry": {
+			SizeStr:    humanize.IBytes(C.DAOS_EC_CELL_MIN),
+			EntryBytes: C.DAOS_EC_CELL_MIN,
+		},
+		"Human Default Entry": {
+			SizeStr:    humanize.IBytes(C.DAOS_EC_CELL_DEF),
+			EntryBytes: C.DAOS_EC_CELL_DEF,
+		},
+		"Human Maximal Entry": {
+			SizeStr:    humanize.IBytes(C.DAOS_EC_CELL_MAX),
+			EntryBytes: C.DAOS_EC_CELL_MAX,
+		},
+		"Bytes Minimal Entry": {
+			SizeStr:    strconv.FormatUint(C.DAOS_EC_CELL_MIN, 10),
+			EntryBytes: C.DAOS_EC_CELL_MIN,
+		},
+		"Bytes Default Entry": {
+			SizeStr:    strconv.FormatUint(C.DAOS_EC_CELL_DEF, 10),
+			EntryBytes: C.DAOS_EC_CELL_DEF,
+		},
+		"Bytes Maximal Entry": {
+			SizeStr:    strconv.FormatUint(C.DAOS_EC_CELL_MAX, 10),
+			EntryBytes: C.DAOS_EC_CELL_MAX,
+		},
+	} {
+		testRunner.Run(testingName, func(testRunner *testing.T) {
+			propEntry := new(C.struct_daos_prop_entry)
+			err := propHdlrs[C.DAOS_PROP_ENTRY_EC_CELL_SZ].nameHdlr(nil,
```

Comment:
On the one hand, I like this as a pragmatic solution to the problem of adding unit tests for input handling. Something is definitely better than nothing! On the other hand, I'm a bit concerned that this specific approach may require a bit too much knowledge on the part of the developer who just wants to add a feature to the daos tool and isn't particularly interested in learning much about how Go unit tests work.

I'm fine with landing this as-is for now, though. I think we'll need to look at adding some unit test helpers to reduce the boilerplate. I was also thinking about how we can add integration testing capabilities for the tool in order to verify that the expected libdaos calls are made in response to various tool inputs, but that's a longer-term goal.